### DISCUSSION TITLE
Worste Nav_PosHold

### DISCUSSION THREAD

**Discussion by ghost** [iNavFlight/inav#7446] [General]
I build up my first copter and after the first flights i can't fix my problem with worse Nav_PosHold solution.
I start in level mode and in around 15m hight I use the swith to activate Nav_AltHold, Nav_PosHold and Heading_Hold in ones. My OSD show >12 satellites and a HDOP <2. For some seconds evrithing look good but the copter start so max bigger and bigger circles around the position like a spinning top or start a fly away. If I deactivate all I can direct take controll back.

Setup
**Board and Sensor Alignment**
Yaw Degrees -> 0,0
MAG Alignment -> CW 270° flip

**GPS configuration**
GPS for navigation and telemetry -> YES
Protocol -> UBLOX7
Ground Assistance Type -> European EGNOS
GPS use Galileo Satellites -> YES
Timezone Offset [Mins] -> 0
Automatic Daylight Saving Time -> OFF

**Comment by avsaase** [comment]
This is called toilet bowling and is caused either by magnetic interference on the magnetometer or incorrect magnetometer alignment (or both). How do you have you magenotometer mounted? Can you provide a photo of your quad?

**Comment by ghost** [comment]
![IMG_20210915_203207 1](https://user-images.githubusercontent.com/90793420/133490068-882396e8-bb9a-4a54-b0da-16cdee9982ee.jpg)
![IMG_20210915_203143 1](https://user-images.githubusercontent.com/90793420/133490169-a85d975d-3b34-4e9b-afd7-a13efc6f3284.jpg)

**Reply by ghost** [reply]
The MAG Alignment -> CW 270° flip is from the manual but for Compass Alignment(Arrow forward and flat mounting). So I think now that I forgot the 180deg yaw rotation. I also have to specify a pitch alignment. But after the compass calibration in the field the shown heading 0, 90, 180, 270 shows fine.

**Reply by avsaase** [reply]
The correct alignment for this orientation is probably
```
set align_mag_roll = x
set align_mag_pitch = 1800
set align_mag_yaw = 900
```
where `x` is the angle of the mag times 10.

However, even if you fix that you may still have problems because your power wires are very close to the mag. The general advice is to keep the mag at least 5 cm away from any magnetic things (notably the motors) and any wires carrying high currents (battery leads and motor wires).

**Reply by ghost** [reply]
In the configuration tab the values for Board and Sensor Alignment (Yaw Degrees -> 0,0, MAG Alignment -> CW 270° flip) I leave them as they are?
The angle of the mag around 12deg bachward had to be a pitch angle offset?

**Reply by ghost** [reply]
Can I check in the sensor tab the correct alignment? what should the values ​​for x, y and z axis be?

**Reply by avsaase** [reply]
You need to put these settings in the CLI. The mag alignment on the configuration tab is then ignored so you can leave it as it is. Don't touch the board yaw setting unless you mounted your flight controller rotated compared to the arrow pointing forward.

**Reply by ghost** [reply]
Thank you very much it works perfekt.
When I fly and stop the forward Pitch the drone fly around 3 to 5 meter backwards. Is there is a possibility to reduce this distance?
